### ISSUE TITLE
Amazon MQ - engine_type validation rule does not correlate with AWS MQ API Behaviour

### ISSUE THREAD

**Issue by Dan-Kitchen** [terraform-linters/tflint#602]
# Overview

`amazon_mq_broker` variable `engine_type` is not being correctly validated by tflint.

Terraform docs state only valid value is "ActiveMQ" - https://www.terraform.io/docs/providers/aws/r/mq_broker.html#engine_type
AWS Docs state - "ACTIVEMQ" - https://docs.aws.amazon.com/amazon-mq/latest/api-reference/brokers-broker-id.html#brokers-broker-id-model-enginetype

When a resource is created using engine_type="ACTIVEMQ" (as per lint rule) the broker is created. However when you run a subsequent plan the set value on the resource is "ActiveMQ" (from the MQ API). This causes a change to be flagged and will keep causing a change loop.

Giving the resource an input of engine_type="ActiveMQ" does not cause this change loop, but causes tflint to fail validation.

Documented approach between AWS Docs & Terraform seems to be out of line. Contrary to documentation it seems AWS MQ API uses "ActiveMQ" but will accept both as valid input! Furthermore, compared to the Terraform AWS provider docs tflint is not validating correctly so it should be amended to match the provider resource/API.

The same issue also applies for `aws_mq_configuration`.

# Example

Example configuration using `engine_type="ACTIVEMQ"`

```hcl
resource "aws_mq_broker" "example" {
  broker_name = "example"

  engine_type        = "ACTIVEMQ"
  engine_version     = "5.15.0"
  host_instance_type = "mq.t2.micro"
  security_groups    = ["<sg_id>"]

  user {
    username = "admin"
    password = "5uper5ecurity"
  }
}
```

**No `tflint` errors thrown**

Terraform state & AWS API after run:

```json
{
      "mode": "managed",
      "type": "aws_mq_broker",
      "name": "example",
      "provider": "provider.aws",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "apply_immediately": false,
            "arn": "arn:aws:mq:eu-west-1:XXXXXXXXX:broker:testing-mq-validation:b-29f15330-f0a2-42da-b4e4-03fce0ddadb8",
            "auto_minor_version_upgrade": false,
            "broker_name": "testing-mq-validation",
            "configuration": [
              {
                "id": "c-07a60ad6-7430-4867-bb64-7dfcef088540",
                "revision": 2
              }
            ],
            "deployment_mode": "SINGLE_INSTANCE",
            "encryption_options": [
              {
                "kms_key_id": "",
                "use_aws_owned_key": true
              }
            ],
            "engine_type": "ActiveMQ",
             ....
}
```

```json
"BrokerName": "testing-mq-validation",
    "BrokerState": "RUNNING",
    "Configurations": {
        "Current": {
            "Id": "c-07a60ad6-7430-4867-bb64-7dfcef088540",
            "Revision": 2
        },
        "History": []
    },
    "Created": "2020-01-29T14:27:04.647Z",
    "DeploymentMode": "SINGLE_INSTANCE",
    "EncryptionOptions": {
        "UseAwsOwnedKey": true
    },
    "EngineType": "ActiveMQ",
```

Subsequent plan output:

```
Terraform will perform the following actions:

  # aws_mq_broker.example must be replaced
-/+ resource "aws_mq_broker" "example" {
        apply_immediately          = false
      ~ arn                        = "arn:aws:mq:eu-west-1:XXXXXXXXX:broker:testing-mq-validation:b-29f15330-f0a2-42da-b4e4-03fce0ddadb8" -> (known after apply)
        auto_minor_version_upgrade = false
        broker_name                = "testing-mq-validation"
        deployment_mode            = "SINGLE_INSTANCE"
      ~ engine_type                = "ActiveMQ" -> "ACTIVEMQ" # forces replacement
        engine_version             = "5.15.9"
```

Example configuration using `engine_type="ActiveMQ"`

```hcl
resource "aws_mq_broker" "example" {
  broker_name = "example"

  engine_type        = "ActiveMQ"
  engine_version     = "5.15.0"
  host_instance_type = "mq.t2.micro"
  security_groups    = ["<sg_id>"]

  user {
    username = "admin"
    password = "5uper5ecurity"
  }
}
```

`tflint` errors:

```
1 issue(s) found:

Error: engine_type is not a valid value (aws_mq_broker_invalid_engine_type)

  on main.tf line 28:
  28:   engine_type        = "ActiveMQ"
```

# Version Info

Terraform v0.12.20
provider.aws v2.45.0
TFLint version 0.13.4

**Comment by Dan-Kitchen** [comment]
Apologies for not reading more earlier but it seems this is backed by the AWS SDK MQ model - https://github.com/aws/aws-sdk-go/blob/master/models/apis/mq/2017-11-27/api-2.json#L1679

This sets the value as "ACTIVEMQ". Seems odd to me as this is different to how the AWS MQ API represents this (regardless of input). It seems the root cause of this issue doesn't lie in tflint...

**Comment by wata727** [comment]
@Dan-Kitchen Thank you for the detailed report. Unfortunately, we know that the enum definition of aws-sdk on which TFLint depends is sometimes wrong. See also https://github.com/terraform-linters/tflint/issues/441

At this point, I think these rules need to be fixed to prevent similar issues.
